src/pybind/mgr/cephadm/ssh.py

```diff
@@ -136,7 +136,13 @@ async def _execute_command(self,
                                addr: Optional[str] = None,
                                ) -> Tuple[str, str, int]:
         conn = await self._remote_connection(host, addr)
-        sudo_prefix = "sudo " if self.mgr.ssh_user != 'root' else ""
+        if self.mgr.ssh_user != 'root':
+                if self.mgr.ssh_password:
```

Comment:
I'm wondering if instead of just checking if a password is set and using it if it's there, we ought to have a mgr module option for whether or not we want to use a sudo password. We have a handful of other type `bool` config options (defined at the top of the `CephadmOrchestrator` class in `pybind/mgr/cephadm/module.py`) that work like this where we take different actions based on whether they're set or not. I think making it more explicit like that could help avoid potential bugs around this by isolating these changes behind having to have set that config option. We could then have the option default to false (which would make this change safer as none of the changes would take effect if it was false) but set it to true ourselves as part of bootstrap if users provide it.